core/vtk/ttkAlgorithm/ttkAlgorithm.h

```diff
@@ -208,6 +208,9 @@ class TTKALGORITHM_EXPORT ttkAlgorithm : public vtkAlgorithm,
 protected:
   ttkAlgorithm();
   ~ttkAlgorithm() override;
+#ifdef TTK_ENABLE_MPI
```

Comment:
I would declare that boolean much higher in the hierarchy (not to exclude the possibility for the future to maybe use mpi outside of paraview/vtk).
I'd recommend to move that to `core/base/common/BaseClass.h`